tokio/src/runtime/basic_scheduler.rs

```diff
@@ -117,13 +138,60 @@ where
         self.spawner.spawn(future)
     }
 
-    pub(crate) fn block_on<F>(&mut self, future: F) -> F::Output
-    where
-        F: Future,
-    {
+    pub(crate) fn block_on<F: Future>(&self, future: F) -> F::Output {
+        // If we can steal the dedicated parker than lets block_on that
+        // otherwise, lets block_on and attempt to steal it back if we can.
+        if let Some(mut inner) = self.take_inner() {
+            inner.block_on(future)
+        } else {
+            // TODO: should this be false or true? In the origina block_on for
+            // basic_scheduler we have false?
+            let enter = crate::runtime::enter(false);
```

Comment:
the bool indicates whether or not blocking calls are allowed in the current context. i am _pretty_ sure that this should remain false.

Reply:
:+1: agreed